Libplanet.Explorer/GraphTypes/CurrencyInputType.cs

```diff
@@ -0,0 +1,108 @@
+            Field<BigIntGraphType>("maximumSupplyMinorUnit");
+            Field<NonNullGraphType<BooleanGraphType>>(
+                "totalSupplyTrackable",
+                "Whether the total supply of this currency is trackable."
+            );
+        }
+
+        public override object ParseDictionary(IDictionary<string, object?> value)
+        {
+            IImmutableSet<Address>? minters = null;
+            var rawMinters = value.TryGetValue("minters", out object? obj) && obj is object[] o
+                ? o
+                : null;
+            if (rawMinters is not null)
+            {
+                if (rawMinters.Any())
+                {
+                    minters = ImmutableHashSet<Address>.Empty;
+                    foreach (var rawMinter in rawMinters)
+                    {
+                        minters = minters.Add((Address)rawMinter);
+                    }
+                }
+            }
+
+            bool totalSupplyTrackable = (bool)value["totalSupplyTrackable"]!;
+            (BigInteger, BigInteger)? maximumSupply = GetMaximumSupply(value);
+            if (maximumSupply is { } && !totalSupplyTrackable)
+            {
+                throw new ExecutionError(
+                    "Maximum supply is not available for legacy untracked currencies.");
+            }
+
+            string ticker = (string)value["ticker"]!;
+            byte decimalPlaces = (byte)value["decimalPlaces"]!;
+
+#pragma warning disable CS0618
+
+            if (!totalSupplyTrackable)
+            {
+                return Currency.Legacy(
+                    ticker,
+                    decimalPlaces,
+                    minters: minters);
+            }
+#pragma warning restore CS0618
+
+            return maximumSupply is { } ms
+                ? Currency.Capped(ticker, decimalPlaces, ms, minters)
+                : Currency.Uncapped(ticker, decimalPlaces, minters);
+        }
+
+        private static (BigInteger, BigInteger)? GetMaximumSupply(
+            IDictionary<string, object?> variables
+        )
+        {
+            BigInteger? nullableMajorUnit =
+                variables.TryGetValue("maximumSupplyMajorUnit", out object? nullableMajorUnitValue)
+                && nullableMajorUnitValue is BigInteger majorUnit
+                    ? majorUnit
+                    : null;
+            BigInteger? nullableMinorUnit =
+                variables.TryGetValue("maximumSupplyMinorUnit", out object? nullableMinorUnitValue)
+                && nullableMinorUnitValue is BigInteger minorUnit
+                    ? minorUnit
+                    : null;
+
+            return (nullableMajorUnit, nullableMinorUnit) switch
+            {
+                (null, null) => null,
+                (BigInteger x, BigInteger y) => (x, y),
+                _ => throw new ExecutionError(
+                    "Both \"maximumSupplyMajorUnit\" and \"maximumSupplyMinorUnit\" must be omitted"
```

Comment:
Maybe:
```suggestion
                    "Both \"maximumSupplyMajorUnit\" and \"maximumSupplyMinorUnit\" must be present or omitted"
```
Since they both would be present or omitted (XNOR in a sense?)